stream-tests/src/test/scala/org/apache/pekko/stream/scaladsl/FlowGroupBySpec.scala

```diff
@@ -694,6 +692,19 @@ class FlowGroupBySpec extends StreamSpec("""
       queue.complete()
     }
 
+    "should not fail when one sub stream completed" in {
+      Source(1 to 10)
+        .groupBy(2, _ % 2, allowClosedSubstreamRecreation = false)
+        .take(2)
```

Comment:
The sub stream is limited to 2, after that, the elements is skiped and keep pulling.